client/src/client.rs

```diff
@@ -187,7 +187,7 @@ impl RawTx for String {
 }
 
 /// The different authentication methods for the client.
-#[derive(Clone, Debug, Hash, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Ord, PartialOrd, Serialize, Deserialize)]
 pub enum Auth {
```

Comment:
I wonder if it would be worth it implementing the serde logic manually because none of the available tagging options feel right. The default, external tagging, would result in:

```json
"auth_field": {
  "UserPass": ["user", "pass"]
}
```

or
 
```json
"auth_field": {
  "CookieFile": "/path/to/file"
}
```

I think we should at least add variant renaming to snake_case. But ideally it would look like this imo:

```json
"auth_field": {
  "user": "user",
  "pass": "pass"
}
```

or
 
```json
"auth_field": {
  "cookie_file": "/path/to/file"
}
```

We could achieve this with untagged serialization and changing the tuple variants to struct variants, but that would be an API break and would make direct usage more cumbersome. We could add constructors to ease the pain, but the API break would remain.